services/powerbi/kubeutils/utils.go

```diff
@@ -0,0 +1,62 @@
+package kubeutils
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os/user"
+	"path/filepath"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// Defines the Kubernetes specific paths
+const (
+	PathKubeNamespace = "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
+)
+
+// CreateKubeClientset creates a new kubernetes clientset
+func CreateKubeClientset() (clientset *kubernetes.Clientset, err error) {
+	var config *rest.Config
+
+	// Always try to get in-cluster config first
+	config, err = rest.InClusterConfig()
+	if err != nil {
+		currentUser, err := user.Current()
+		if err != nil {
+			return nil, fmt.Errorf("Fail to get the current user")
+		}
+
+		kubeconfigPath := filepath.Join(currentUser.HomeDir, ".kube", "config")
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+		if err != nil {
+			return nil, fmt.Errorf("Fail to create kubernetes config from %s | Error %s", kubeconfigPath, err)
+		}
+	}
+
+	clientset, err = kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("Fail to create kubernetes client | Error %s", err)
+	}
+
+	return
+}
+
+// GetCurrentNamespace returns the namespace this Pod belongs to. If it fails
+// to resolve the name, it uses the fallback name.
+func GetCurrentNamespace(fallback string) string {
```

Comment:
Can we shall all these across services and agents?

Reply:
Yesss!

Reply:
Fixed